### DISCUSSION TITLE
What is the option --exact-timestamp in s3 sync good for, if timestamps can't be set and are not preserved?

### DISCUSSION THREAD

**Discussion by apacha** [aws/aws-cli#7071] [Q&A] [answered]
When you are synchronizing files between two S3 buckets, it seems like the modified timestamp of files can't be set and will [not be preserved when copying files](https://github.com/aws/aws-cli/issues/3415#issuecomment-401832723). At least I didn't find any option that would allow me to do this.

Now assuming the following scenario:

```
bucket_A                       bucket_B
File1 (timestamp 1)            -
```

after running `aws s3 sync s3://bucket_A s3://bucket_B` it will be 

```
bucket_A                       bucket_B
File1 (timestamp 1)            File1 (timestamp 2)
```

if I copy files back with `aws s3 sync s3://bucket_B s3://bucket_A` it will be 

```
bucket_A                       bucket_B
File1 (timestamp 3)            File1 (timestamp 2)
```

In which scenario would it EVER be useful to specify the `--exact-timestamp` option, if you can't preserve modified timestamps, thus will ALWAYS have different timestamps for the same files in different buckets? 

Bottom line: I would like to have an option to preserve timestamps - then it would make sense to have this parameter, otherwise it seems to be completely useless.

**Comment by aBurmeseDev** [comment, marked as answer]
Hi @apacha, thanks for reaching out. This topic has come up a few times before in other issues. I think [this comment](https://github.com/aws/aws-cli/issues/2000#issuecomment-226299644) addresses your question:

> The `--exact-timestamps` is not for setting the LastModifiedTime of the S3 objects. This is not possible to do via the S3 api...In other words, this is only used to control what files are synced, not to sync the timestamps.

Also [this comment](https://github.com/aws/aws-cli/issues/6601#issuecomment-990137502) from a related issue provides more context as well. As mentioned there, you could pass the last modified time as metadata. Hopefully that helps, please let us know if you have any feedback on how the [documentation](https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html) for this could be improved.
